src/Containers-SmallOrderedSet/CTSmallOrderedSetTest.class.st

```diff
@@ -81,14 +82,242 @@ CTSmallOrderedSetTest >> testDo [
 	self assert: (seen at: 2) equals: 1
 ]
 
-{ #category : #testing }
+{ #category : 'testing' }
+CTSmallOrderedSetTest >> testDoSeparatedBy [ 
+    | seen |
+
+    "Add elements manually, including duplicates"
+    collection add: 2.
+    collection add: 1.
+    collection add: 3.
+    collection add: 2.  "Duplicate"
+    collection add: 1.  "Duplicate"
+
+    "Verify that collection size is correct (duplicates should not be counted)"
```

Comment:
Can you split this test in two?
One test testDuplicatedObjectsAreNotAdded
One test testDoSeparatedBy